File: app/src/main/java/com/rbppl/to_do_list/data/TaskDao.java
package com.rbppl.to_do_list.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import androidx.room.Delete;

import java.util.List;

@Dao
public interface TaskDao {

    @Query("SELECT * FROM task")
    public LiveData<List<Task>> getAllTasks();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public long insert(Task task);

    @Update
    public int update(Task task);

    @Delete
    public int delete(Task task);
}

File: app/src/main/java/com/rbppl/to_do_list/data/TaskDao.kt
package com.rbppl.to_do_list.data
import androidx.lifecycle.LiveData
import androidx.room.*
@Dao
interface TaskDao {
    @Query("SELECT * FROM task")
    fun getAllTasks(): LiveData<List<Task>>
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(task: Task): Long
    @Update
    suspend fun update(task: Task): Int
    @Delete
    suspend fun delete(task: Task): Int
}
